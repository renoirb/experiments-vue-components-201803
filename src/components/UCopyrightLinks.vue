<template>
  <div
    :class="[classes]"
  >
    <a
      :href="productUrl"
    >
      {{ productName }}
    </a>
    &nbsp;
    &copy;
    &nbsp;
    {{ firstYear }}-{{ year }}
    &nbsp;
    <a
      :href="publisherUrl"
    >
      {{ publisherName }}
    </a>
  </div>
</template>

<script>
import Themeable from 'vuetify/es5/mixins/themeable'
import Colorable from 'vuetify/es5/mixins/colorable'

// @vue/component
export default {
  name: 'u-copyright-links',
  mixins: [
    Colorable,
    Themeable,
  ],
  props: {
    productName: {
      type: String,
      default: 'CatFacts',
    },
    productUrl: {
      type: String,
      default: 'http://www.catfacts.co/',
    },
    publisherName: {
      type: String,
      default: 'PiedPiper',
    },
    publisherUrl: {
      type: String,
      default: 'http://www.piedpiper.com/',
    },
    firstYear: {
      type: Number,
      default: 1998,
    },
  },
  computed: {
    year () {
      return new Date().getFullYear()
    },
    classes () {
      /**
       * Here, we're using Vuetify's mixins.
       * So we can see how to leverage them
       * in our own components.
       */
      const classes = Object.assign(
        {
          'u-copyright-links--component': true,
          'text-xs-center': true,
          'pa-1': true,
          'accent--text': true,
          'theme--dark': this.dark,
          'theme--light': this.light,
        },
        this.color
          ? Colorable.methods.addTextColorClassChecks.call(this, {}, this.color)
          : {}
      )

      return classes
    },
  },
}
</script>
