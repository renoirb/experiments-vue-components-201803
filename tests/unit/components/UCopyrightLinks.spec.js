import {
  shallow,
} from '@vue/test-utils'
import Vue from 'vue'
import UCopyrightLinks from '@/components/UCopyrightLinks.vue'

describe('UCopyrightLinks.vue', () => {
  it('renders props.productName when passed', () => {
    const productName = 'sg59PgYRUqI79HW'
    const wrapper = shallow(UCopyrightLinks, {
      propsData: {
        productName,
      },
    })
    expect(wrapper.text()).toMatch(productName)
    expect(wrapper.element).toMatchSnapshot()
  })
  it('Should have the following default markup', () => {
    // Thanks  https://alexjoverm.github.io/2017/08/21/Write-the-first-Vue-js-Component-Unit-Test-in-Jest/
    let Component = Vue.extend(UCopyrightLinks) // Create a copy of the original component
    const vm = new Component({
      propsData: { // Replace data value with this fake data
        productName: 'SomethingElse',
      },
    }).$mount() // Instances and mounts the component
    expect(vm.$el).toMatchSnapshot()
  })
})
