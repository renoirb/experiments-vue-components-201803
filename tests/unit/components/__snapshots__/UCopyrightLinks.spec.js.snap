// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`u-copyright-links Should have the following default markup 1`] = `
<div
  class="u-copyright-links--component text-xs-center pa-1 accent--text"
>
  <a
    href="http://www.catfacts.co/"
  >
    
    SomethingElse
  
  </a>
  
   
  ©
   
  1998-2018
   
  
  <a
    href="http://www.piedpiper.com/"
  >
    
    PiedPiper
  
  </a>
</div>
`;

exports[`u-copyright-links renders props.productName when passed 1`] = `
<div
  class="u-copyright-links--component text-xs-center pa-1 accent--text"
>
  <a
    href="http://www.catfacts.co/"
  >
    
    sg59PgYRUqI79HW
  
  </a>
  
   
  ©
   
  1998-2018
   
  
  <a
    href="http://www.piedpiper.com/"
  >
    
    PiedPiper
  
  </a>
</div>
`;
